build/common.sh

```diff
@@ -793,7 +828,14 @@ function kube::release::create_docker_images_for_server() {
         ln ${binary_dir}/${binary_name} ${docker_build_path}/${binary_name}
         printf " FROM ${base_image} \n ADD ${binary_name} /usr/local/bin/${binary_name}\n" > ${docker_file_path}
 
-        local docker_image_tag=gcr.io/google_containers/$binary_name:$md5_sum
+        if [[ ${arch} == "amd64" ]]; then
```

Comment:
At least something here: https://github.com/kubernetes/kubernetes/blob/master/cluster/common.sh#L400

and maybe related to https://github.com/kubernetes/kubernetes/blob/master/cluster/saltbase/install.sh#L27

I think there are more places too, but can we take that in a separate PR or something?
To adopt `-amd64` naming everywhere is separate from building for other arches